src/conditions/mod.rs

```diff
@@ -5,7 +5,17 @@ use inventory;
 pub mod check_fields;
 
 pub trait Condition: Send + Sync {
-    fn check(&self, e: &Event) -> bool; // TODO: Add method that provides fail context? -> Result<(), String>
+    fn check(&self, e: &Event) -> bool;
+
+    // Provides context for a failure. This is potentially mildly expensive if
```

Comment:
There's no harm in doc-commenting useful private functions. It is useful for IDE tooling.